### ISSUE TITLE
OverflowMenu steals back focus when clicking outside and it collapses

### ISSUE THREAD

**Issue by scottdickerson** [carbon-design-system/carbon#6227]
## What package(s) are you using?

- [ ] `carbon-components`
- [x] `carbon-components-react`

## Detailed description

> Describe in detail the issue you're having.

If I click on a button (or really anything) while the OverflowMenu is expanded.  It steals focus back to the collapsed Menu button instead of focusing on what I clicked on

> Is this issue related to a specific component?

> What did you expect to happen? What happened instead? What would you like to
> see changed?

> What browser are you working in?

> What version of the Carbon Design System are you using?

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

## Steps to reproduce the issue

Take a look at the sandbox below.
Expand the Overflow Menu
Now click on the button next to the Overflow Menu.
Notice that the OverflowMenu Icon has stolen the focus back to itself

> Please create a reduced test case in CodeSandbox

https://codesandbox.io/s/carbon-overflow-menu-focus-stealing-9k1pl

## Additional information

- Screenshots or code
Here's a picture of the stolen focus after I clicked the Click me button
![image](https://user-images.githubusercontent.com/6663002/84170793-a5d3bc80-aa3f-11ea-9f8b-2a9314486b59.png)

**Comment by emyarod** [comment]
IIRC this was intentional due to a11y requirements, but unfortunately the original issue was wiped from the old issue tracker (my only point of reference is the related commit  https://github.com/carbon-design-system/carbon-components-react/commit/5dc69b9d7a427531c8449381851e21dfe9e175c0)

**Comment by dakahn** [comment]
I initially thought this was a misreport (in no small part due to Codesandbox being absolutely unreliable on Firefox/Chrome on my Windows 10 machine), but now I see the bug. The APG spec referenced is [_I think this one_](https://www.w3.org/TR/wai-aria-practices-1.1/#keyboard-interaction-12)

And correctly, if the menu is open and the user hits escape -- focus returns to the triggering element, **but** we also do this on blur so, as described in the issue -- if the user clicks a text input or button, focus does not move to those elements, but instead moves to the triggering element. 

Removing this functionality on blur makes sense to me unless there's something I'm missing.

**Comment by gptt916** [comment]
Does this issue still need to be fixed? I would like to give it a go if no one is currently working on it.

**Comment by tw15egan** [comment]
@gptt916 I don't think anyone is actively working on this so if you'd like to work on it that would be very much welcomed!
